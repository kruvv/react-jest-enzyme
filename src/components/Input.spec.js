import { shallow } from 'enzyme';
import React from 'react';
import Input from './Input';

describe('<Input />', () => {
    it('should correctly render component Input', () => {
        const component = shallow(<Input />);
        expect(component).toMatchSnapshot();
    });

    describe("defaultProps", () => {
        it('should use default onChange', () => {
            const result = Input.defaultProps.onChange();
            expect(result).toBeUndefined();
        });
        it('should use default onKeyPress', () => {
            const result = Input.defaultProps.onKeyPress();
            expect(result).toBeUndefined();
        });
    });

});
