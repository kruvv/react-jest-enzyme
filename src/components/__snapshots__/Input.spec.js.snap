// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should correctly render component Input 1`] = `
<div
  className="inputWrapper"
>
  <i
    className="fas fa-search"
  />
  <input
    className="input"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Click me for search"
    value=""
  />
</div>
`;
